reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServerConfig.java

```diff
@@ -819,7 +832,24 @@ public void onChannelInit(ConnectionObserver observer, Channel channel, @Nullabl
 			boolean needRead = false;
 
 			if (sslProvider != null) {
-				sslProvider.addSslHandler(channel, remoteAddress, SSL_DEBUG);
+				NonSslRedirectHandler nonSslRedirectHandler = null;
+				ChannelPipeline pipeline = channel.pipeline();
+				if (redirectHttp2Https) {
+					nonSslRedirectHandler = new NonSslRedirectHandler();
+					NonSslRedirectDetector nonSslRedirectDetector = new NonSslRedirectDetector(nonSslRedirectHandler,
+							sslProvider,
+							remoteAddress,
+							SSL_DEBUG);
+					if (pipeline.get(NettyPipeline.ProxyHandler) != null) {
```

Comment:
Check `NettyPipeline` javadoc. `ProxyHandler` is applied only when we are on the client, so we do not need to check for this handler on the server.

Reply:
Thanks. I will remove the check.